src/sentry/snuba/discover.py

```diff
@@ -1325,6 +1325,69 @@ def get_facets(
     return results
 
 
+def spans_histogram_query(
+    span,
+    user_query,
+    params,
+    num_buckets,
+    precision=0,
+    min_value=None,
+    max_value=None,
+    data_filter=None,
+    referrer=None,
+    group_by=None,
+    order_by=None,
+    limit_by=None,
+    extra_snql_condition=None,
+    normalize_results=True,
+):
+    multiplier = int(10 ** precision)
+    if max_value is not None:
+        max_value -= 0.1 / multiplier
+
+    # TODO add min max calculation after
+    # min_value, max_value = find_histogram_min_max(
+    #     fields, min_value, max_value, user_query, params, data_filter, use_snql
+    # )
```

Comment:
You should make a new ticket to directly follow up this PR and do these min/max next this since it's required to dynamically create buckets below.